packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -363,6 +372,12 @@ export async function enqueueLinks(
 
     let requestOptions = createRequestOptions(urls, options);
 
+    if (robotsFile) {
+        requestOptions = requestOptions.filter((request) => {
+            return robotsFile.isAllowed(request.url);
+        });
+    }
+
```

Comment:
If it is simple, I would also support this filter in `crawler.addRequests`. I know it is a small wrapper above `requestQueue.addRequests` but since it is on the `crawler` object, users will expect it will respect robots. It would drop those requests later when fetching but polluting and draining the queue is bad for performance.

Reply:
Right, I was thinking about that one as well, it should be simple, will do.

Reply:
implemented via e230191

btw this is just a perf optimization, technically, it was already working this way, since we check if the request is valid inside the `_runTaskFunction`. now we also skip the disallowed ones from adding to the queue via `crawler.addRequests` like we do with `enqueueLinks`.

Reply:
> like we do with `enqueueLinks`

Note that for `enqueueLinks`, we'll only check against the current sitemap (possibly enqueueing forbidden different-domain links), but with `addRequests`, we'll check the `robots.txt` files for all of the links separately (possibly downloading many `robots.txt` files).

It kinda makes sense to me (and as you're saying, it's just a matter of performance), just making sure we all understand this right.

Reply:
(nvm, the performance difference is just RQ utilization, the requests to `robots.txt` files will be made either way)